File: BOOM/TechnicalServices/SystemRandomGenerator.cs
using System;

namespace TechnicalServices
{
    //Implementation of IRandomGenerator that uses the System.Random functionality
    //to produce random numbers.
    public class SystemRandomGenerator : IRandomGenerator
    {
        private static Random Random = new Random();

        public int Next(int minValue, int maxValue)
        {
            return Random.Next(minValue, maxValue);
        }
    }
}

File: BOOM.VB/TechnicalServices/SystemRandomGenerator.vb


Namespace TechnicalServices
	'Implementation of IRandomGenerator that uses the System.Random functionality
	'to produce random numbers.
	Public Class SystemRandomGenerator
		Implements IRandomGenerator
		Private Shared Random As New Random()

        Public Function NextValue(minValue As Integer, maxValue As Integer) As Integer Implements IRandomGenerator.NextValue
            Return Random.Next(minValue, maxValue)
        End Function
    End Class
End Namespace

